### PR TITLE
Repos model

### PR DESCRIPTION
**Pull request by alfredodeza**
Implements the model for Repos, adding tests too.

Moves the listener for modified timestamps to a generic spot.

### CODE REVIEW

File: chacra/models/binaries.py [hunk 1]
```diff
@@ -28,6 +29,9 @@ class Binary(Base):
     project_id = Column(Integer, ForeignKey('projects.id'))
     project = relationship('Project', backref=backref('binaries', lazy='dynamic'))
 
+    repo_id = Column(Integer, ForeignKey('repos.id'))
```
Comment (andrewschoen) [inline chacra/models/binaries.py]: This needs to be a many to many relationship, right?  Binaries can be in multiple repos and repos can have multiple binaries.

http://docs.sqlalchemy.org/en/improve_toc/orm/basic_relationships.html#many-to-many

File: chacra/models/repos.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    path = Column(String(256))
+    ref = Column(String(256), index=True)
+    distro = Column(String(256), nullable=False, index=True)
+    distro_version = Column(String(256), nullable=False, index=True)
+    modified = Column(DateTime, index=True)
+    signed = Column(Boolean(), default=False)
+    needs_update = Column(Boolean(), default=False)
+    size = Column(Integer, default=0)
+
+    binary_id = Column(Integer, ForeignKey('projects.id'))
```
Comment (andrewschoen) [inline chacra/models/repos.py]: you've called this ``binary_id`` and linked to ``projects.id``. should this be project_id instead?

File: chacra/models/repos.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import datetime
+from sqlalchemy import Column, Integer, String, ForeignKey, Boolean, DateTime
+from sqlalchemy.orm import relationship, backref
+from sqlalchemy.event import listen
+from sqlalchemy.orm.exc import DetachedInstanceError
+from chacra.models import Base, update_timestamp
+
+
+class Repo(Base):
```
Comment (andrewschoen) [inline chacra/models/repos.py]: Did you forget the ``needs_update`` flag or decide against it?

File: chacra/models/repos.py (hunk 2 above)
Comment (alfredodeza) [inline chacra/models/repos.py]: hah, good catch. Fixed.

File: chacra/models/repos.py (hunk 3 above)
Comment (alfredodeza) [inline chacra/models/repos.py]: I forgot initially but added it last, you don't see it?

Comment (andrewschoen) [inline chacra/models/repos.py]: I see it now

File: chacra/models/binaries.py (hunk 1 above)
Comment (andrewschoen) [inline chacra/models/binaries.py]: This was discussed in IRC, the m2m relationships will be configured instead of having a db relationship.

**Pull request merged**